pkg/kubectl/cmd/create_clusterrole.go

```diff
@@ -133,6 +134,21 @@ func (c *CreateClusterRoleOptions) Validate() error {
 				return fmt.Errorf("invalid verb: '%s' for nonResourceURL", v)
 			}
 		}
+
+		for _, nonResourceURL := range c.NonResourceURLs {
+			if nonResourceURL == "*" {
+				continue
+			}
+
+			if !strings.HasPrefix(nonResourceURL, "/") {
+				return fmt.Errorf("nonResourceURL should start with /")
+			}
+
+			count := strings.Count(nonResourceURL, "*")
+			if count > 1 || (count == 1 && !strings.HasSuffix(nonResourceURL, "*")) {
```

Comment:
nit: a better way of expressing line 147,148 would be
strings.ContainsRune(url[:len(url)-1], '*')